File: server_client_java_protoV3/JavaSocketsFestival_proto/Java11ChatSockets/ChatPersistence/src/main/java/festival/persistence/repository/DBRepoAngajat.java
package festival.persistence.repository;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import festival.model.Angajat;
import festival.model.Bilet;
import festival.model.Identifier;
import festival.model.Persoana;

import java.sql.*;
import java.util.*;

import java.util.Optional;

//import org.mindrot.jbcrypt.BCrypt;

import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;


public class DBRepoAngajat <ID,E extends Identifier<ID>> implements IDBRepoAngajat{

    private JdbcUtils dbUtils;
    private static final Logger logger= LogManager.getLogger();
    private static final BCryptPasswordEncoder encoder = new BCryptPasswordEncoder();


    public DBRepoAngajat(Properties props) {
        System.out.println("Initializing DBRepoAngajat with properties: ");
        logger.info("Initializing DBRepoAngajat with properties: {} ",props);
        dbUtils=new JdbcUtils(props);
    }


    @Override
    public Iterable findAll() {
        return null;
    }

    @Override
    public Optional save(Identifier entity) {

        return Optional.empty();
    }

    @Override
    public Optional findOne(Object longID) {
        return Optional.empty();
    }

    @Override
    public Optional delete(Object aLong) {
        return Optional.empty();
    }


//    private String hashPassword(String password) {
//        return BCrypt.hashpw(password, BCrypt.gensalt());
//    }
//
//    // Metoda pentru verificarea parolei
//    private boolean verifyPassword(String candidate, String hashedPassword) {
//        return BCrypt.checkpw(candidate, hashedPassword);
//    }

    // Metoda pentru criptarea parolei
    public String hashPassword(String password) {
        return encoder.encode(password);
//
//        return password;
    }

    //     Metoda pentru verificarea parolei
    public static boolean verifyPassword(String candidate, String hashedPassword) {
        return encoder.matches(candidate, hashedPassword);
//        return false;
    }


    // Metoda pentru autentificare
    public Angajat authenticate(String u, String pass) {
       // hashPassword("Mirel");
//        hashPassword("Dana");
        System.out.println("repo autentificate: "+u);

        logger.traceEntry();
        Connection con = dbUtils.getConnection();
//        List<Bilet> Bilete = new ArrayList<>();
        try (PreparedStatement preStmt = con.prepareStatement(
                "select * from persoana P JOIN angajat A ON A.IDPersoana=P.IDPerosana where A.username = ?" ))

        {

            preStmt.setString(1, u);
            ResultSet resultSet = preStmt.executeQuery();
            if (resultSet.next()) {
                String hashedPassword = resultSet.getString("password"); // coloana pentru parola criptată
                if (verifyPassword(pass, hashedPassword)) {
                    String firstName = resultSet.getString("nume");
                    String lastName = resultSet.getString("prenume");
                    String oficiu = resultSet.getString("oficiu");
                    Long id_ang = Long.valueOf(resultSet.getString("IDAngajat"));

                    Angajat ut = new Angajat(id_ang,firstName, lastName);
                    ut.setOficiu(oficiu);
                    ut.setPassword(pass);
                    ut.setUsername(u);
                    System.out.println(ut.getUsername());

                    return ut;
                }
//                else
//                { try (PreparedStatement updateStmt = con.prepareStatement(
//                        "UPDATE angajat SET password = ? WHERE username = ?")) {
//                                    pass=hashPassword(pass);
//
//                    updateStmt.setString(1, pass);
//                    updateStmt.setString(2, u);
//                    updateStmt.executeUpdate();
//                }
//                }
            }
        } catch (SQLException e) {
            throw new RuntimeException(e);
        }

        return null;
    }
}

File: server_client_C_sharp/mpp_c_app1/FestivalPersistence/DBRepoAngajat.cs
using System;
using System.Collections.Generic;
using System.Data.SqlClient;
using System.Data.SQLite;
using log4net;
//using Mysqlx.Crud;


using System;
using System.Security.Cryptography;
using System.Text;
using FestivalModel;

namespace FestivalPersistence
{


    public class PasswordHasher
    {
        // Encoderul pentru hash
        private static readonly SHA256 sha256 = SHA256.Create();

        // Metoda pentru criptarea parolei
        public string HashPassword(string password)
        {
            byte[] hashedBytes = sha256.ComputeHash(Encoding.UTF8.GetBytes(password));
            return Convert.ToBase64String(hashedBytes);
        }

        // Metoda pentru verificarea parolei
        public bool VerifyPassword(string candidate, string hashedPassword)
        {
            string candidateHash = HashPassword(candidate);
            return candidateHash == hashedPassword;
        }
    }


    public class AuthenticationService
    {
        private PasswordHasher hasher;
        private SQLiteConnection connection;

        public AuthenticationService(string connectionString)
        {
            hasher = new PasswordHasher();
            connection = new SQLiteConnection(connectionString);
        }

        public void Update(string newHashedPassword, string u)
        {
            using (connection)
            {
                connection.Open();
                string query1 = "UPDATE angajat SET password = @newHashedPassword WHERE username = @username";
                using (SQLiteCommand command1 = new SQLiteCommand(query1, connection))
                {
                    command1.Parameters.AddWithValue("@newHashedPassword", newHashedPassword);
                    command1.Parameters.AddWithValue("@username", u);
                    command1.ExecuteNonQuery();
                }

                connection.Close();
            }
        }

        // Metoda pentru autentificare
        public Angajat Authenticate(string u, string pass)
        {
            string query =
                "select * from persoana P JOIN angajat A ON A.IDPersoana=P.IDPerosana where A.username = @username";
            using (SQLiteCommand command = new SQLiteCommand(query, connection))
            {
                command.Parameters.AddWithValue("@username", u);
                connection.Open();
                using (SQLiteDataReader reader = command.ExecuteReader())
                {
                    while (reader.Read())
                    {
                        string hashedPassword = reader["password"].ToString();
                        if (hasher.VerifyPassword(pass, hashedPassword))
                        {
                            // Dacă parola este corectă, creează și întoarce obiectul Angajat
                            string firstName = reader["nume"].ToString();
                            string lastName = reader["prenume"].ToString();
                            string office = reader["oficiu"].ToString();
                            long id_ang = Convert.ToInt32(reader["IDAngajat"]);

                            Angajat angajat = new Angajat(id_ang, firstName, lastName);
                            angajat.IdPersoana = id_ang;
                            angajat.Oficiu = office;
                            connection.Close();

                            return angajat;
                        }
                    }
                }
            }

            // Dacă nu s-a găsit un utilizator cu acel username sau parola este incorectă, întoarce null
            connection.Close();
            return null;
        }
    }



    public interface IDBRepoAngajat : IRepository<long, Angajat>
    {
    }

    public class DBRepoAngajat : IDBRepoAngajat
    {

        private SQLiteConnection connection;
        private static readonly ILog logger = LogManager.GetLogger("DBRepoArtist");

        public DBRepoAngajat(string connectionString)
        {
            logger.Info("Initializing DBRepoArtist...");
            connection = new SQLiteConnection(connectionString);
        }

        public Angajat findOne(long id)
        {
            throw new NotImplementedException();
        }

        public IEnumerable<Angajat> findAll()
        {
            throw new NotImplementedException();
        }

        public void save(Angajat entity)
        {
            throw new NotImplementedException();
        }

        public void delete(long id)
        {
            throw new NotImplementedException();
        }



    }


}